packages/tailwind/src/utils/react/is-component.ts

```diff
@@ -0,0 +1,9 @@
+export const isComponent = (
+  element: React.ReactElement,
+): element is React.ReactElement<unknown, React.FC<unknown>> => {
+  return (
+    typeof element.type === "function" ||
+    // @ts-expect-error - we know this is a component that may have a render function
```

Comment:
what's the error you get if you don't include the expect error? it would be nice not to use it, perhaps some better assertion with type casting can already help

Reply:
Initially I tried defining a few lines of code to get around this, but the amount of code that's going to need changing is a lot compared to get around the expect error. 😞

The issue is that React doesn't include the `render` in their types as it isn't something for user land, mostly.